Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1144 @@
+        definitive outcome.
+
+        Args:
+            job (dict): The job object that has just completed polling.
+            last_poll_response (list): The raw response from the final polling command.
+
+        Returns:
+            list[QuarantineResult]: A list of final QuarantineResult objects.
+        """
+        demisto.debug(f"[{self.brand} Handler] Finalizing job.")
+        final_results = []
+
+        outputs = Command.get_first_filled_entry_context_list(last_poll_response)
+        demisto.debug(f"[{self.brand} Handler] Finalizing {len(outputs)} endpoint results from job.")
+        for res in outputs:
+            try:
+                final_results.append(self._process_final_endpoint_status(res, job))
+            except Exception as e:
+                demisto.error(
+                    f"[{self.brand} Handler] Failed to get status of quarantine for endpoint {res.get('endpoint_id')}: {e}"
+                )
+                final_results.append(
+                    QuarantineResult(
+                        endpoint_id=res.get("endpoint_id", "Unknown"),
+                        status=QuarantineResult.Statuses.FAILED,
+                        message=QuarantineResult.Messages.GENERAL_FAILURE,
+                        brand=self.brand,
+                        file_path=job.get("finalize_args", {}).get("file_path", ""),
+                        file_hash=job.get("finalize_args", {}).get("file_hash", ""),
+                    )
+                )
+        return final_results
+
+
+def handler_factory(brand: str, orchestrator) -> BrandHandler:
+    """
+    Factory function that returns an instance of the correct brand handler.
+
+    This allows the orchestrator to dynamically select the appropriate logic
+    based on the brand name discovered for a group of endpoints.
+
+    Args:
+        brand (str): The name of the brand.
+        orchestrator (QuarantineOrchestrator): The orchestrator instance.
+
+    Returns:
+        BrandHandler: An instance of a concrete BrandHandler subclass (e.g., XDRHandler).
+
+    Raises:
+        ValueError: If no handler is available for the specified brand.
+    """
+    demisto.debug(f"[Factory] Creating handler for brand: '{brand}'")
+    if brand in [Brands.CORTEX_CORE_IR, Brands.CORTEX_XDR_IR]:
+        demisto.debug("[Factory] Selected XDRHandler.")
+        return XDRHandler(brand, orchestrator)
+    raise ValueError(f"No handler available for brand: {brand}")
+
+
+""" SCRIPT ORCHESTRATOR """
+
+
+class QuarantineOrchestrator:
+    """Manages the entire quarantine lifecycle from start to finish."""
+
+    CONTEXT_PENDING_JOBS = "quarantine_pending_jobs"
+    CONTEXT_COMPLETED_RESULTS = "quarantine_completed_results"
+    ENDPOINT_IDS_ARG = "endpoint_id"
+    FILE_HASH_ARG = "file_hash"
+    FILE_PATH_ARG = "file_path"
+    BRANDS_ARG = "brands"
+
+    HASH_TYPE_TO_BRANDS = {"sha256": [Brands.CORTEX_CORE_IR, Brands.CORTEX_XDR_IR]}
```

Comment:
didn't find anything related, is there a way we distinct between the two integrations according to the tenant? (xsiam/xsoar)